connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/WorkerErrantRecordReporter.java

```diff
@@ -99,8 +102,15 @@ public WorkerErrantRecordReporter(
                 valLength, key, value, headers);
         }
 
-        Future<Void> future = retryWithToleranceOperator.executeFailed(Stage.TASK_PUT,
-            SinkTask.class, consumerRecord, error);
+        Future<Void> future;
+        try {
+            future = retryWithToleranceOperator.executeFailed(Stage.TASK_PUT,
+                SinkTask.class, consumerRecord, error);
+        } catch (ConnectException e) {
+            mustThrowException = true;
+            exceptionToThrow = e;
+            throw e;
+        }
```

Comment:
we don't need these vars, the errors are already stored in the ProcessingContext. look at comment above.